File: SDKs/defold/intelliversex/discord_moderation.lua
-- Copyright (c) 2026 Intelli-verse-X
-- MIT License — see LICENSE in the project root.

--- Discord moderation — stub matching Unity IVXDiscordModeration.
--- @module discord_moderation

local M = {}

M.auto_moderate_enabled = true

function M.enable_auto_moderation(_enable)
    error("Not implemented")
end

function M.process_moderation_metadata(_message_id, _metadata)
    error("Not implemented")
end

function M.get_moderation_action(_metadata)
    error("Not implemented")
end

function M.start_voice_moderation_capture(_lobby_id)
    error("Not implemented")
end

function M.stop_voice_moderation_capture()
    error("Not implemented")
end

function M.report_user(_user_id, _reason)
    error("Not implemented")
end

return M

File: SDKs/web3/src/discord-moderation.ts
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

export enum IVXModerationAction {
  Show = 'Show',
  Hide = 'Hide',
  Blur = 'Blur',
  Replace = 'Replace',
}

/** Parsed moderation outcome for a Discord message. */
export interface IVXModerationDecision {
  messageId: string;
  action: IVXModerationAction;
  reason: string;
  replacement: string;
  severity: string;
}

/**
 * Discord moderation metadata, voice capture for moderation pipelines, and user reporting.
 * Stub surface matching Unity IVXDiscordModeration.
 */
export class IVXDiscordModeration {
  autoModerateEnabled = true;

  enableAutoModeration(_enable: boolean): void {
    throw new Error('Not implemented');
  }

  processModerationMetadata(
    _messageId: string,
    _metadata: Record<string, string>
  ): void {
    throw new Error('Not implemented');
  }

  static getModerationAction(
    _metadata: Record<string, string> | null
  ): IVXModerationDecision {
    throw new Error('Not implemented');
  }

  startVoiceModerationCapture(_lobbyId: string): void {
    throw new Error('Not implemented');
  }

  stopVoiceModerationCapture(): void {
    throw new Error('Not implemented');
  }

  async reportUser(_userId: string, _reason: string): Promise<boolean> {
    throw new Error('Not implemented');
  }
}
